lib/cartopy/_trace.cpp

```diff
@@ -558,7 +563,7 @@ void _project_segment(GEOSContextHandle_t handle,
     t_current = 0.0;
     state = get_state(p_current, gp_domain, handle);
 
-    while(t_current < 1.0)
+    while(t_current < 1.0 && lines.size() < 100)
```

Comment:
Thanks @QuLogic. And thanks for being very specific about this being *lines*. I'm trying to come up with a case where I would want to split a line into more than 100 components and can't (at a push, perhaps highly interrupted projections, or a track which crosses the dateline many times). Let's play it safe, and go for 500.